src/NHibernate/PropertyNotFoundException.cs

```diff
@@ -67,11 +75,28 @@ public PropertyNotFoundException(string propertyName, string fieldName, System.T
 		/// </param>
 		protected PropertyNotFoundException(SerializationInfo info, StreamingContext context) : base(info, context)
 		{
+			this.targetTypeAssemblyQualifiedName = info.GetString("targetTypeAssemblyQualifiedName");
+			this.propertyName = info.GetString("propertyName");
+			this.accessorType = info.GetString("accessorType");
+		}
+
+		[SecurityCritical]
+		public override void GetObjectData(SerializationInfo info, StreamingContext context)
+		{
+			base.GetObjectData(info, context);
+			info.AddValue("targetTypeAssemblyQualifiedName", targetTypeAssemblyQualifiedName);
+			info.AddValue("propertyName", propertyName);
+			info.AddValue("accessorType", accessorType);
 		}
 
 		public System.Type TargetType
 		{
-			get { return targetType; }
+			get
+			{
+				if (targetType != null || targetTypeAssemblyQualifiedName == null) return targetType;
```

Comment:
From #1425 review:

Why not restoring it in the deserialization constructor? (It will allow keeping it `readonly` by the way. The additional field could be avoided too.)